CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+cmake_minimum_required(VERSION 3.8.1)
+project(quick LANGUAGES NONE)
+
+
+if(NOT INSIDE_AMBER)
+
+	# initialization and include paths
+	#-------------------------------------------------------------
+	include(cmake/AmberBuildSystemInit.cmake NO_POLICY_SCOPE)
+
+	#now enable the languages
+	enable_language(C CXX Fortran)
+
+	include(AmberBuildSystem2ndInit)
+	
+	include(MPIConfig)
+	include(CudaConfig)
+
+	set(LINALG_LIBS_REQUIRED TRUE)
+	set(NEEDED_3RDPARTY_TOOLS blas lapack mkl)
+	set(REQUIRED_3RDPARTY_TOOLS )
+	set(BUNDLED_3RDPARTY_TOOLS blas lapack)
+	include(3rdPartyTools)
+	
+	include(CompilerFlags)
+
+	
+	#CPack setup
+	# --------------------------------------------------------------------
+	set(PACKAGE_NAME "quick")
+	set(PACKAGE_FILENAME "q")
+
+	set(BUNDLE_IDENTIFIER org.merzlab.quick)
+	set(BUNDLE_SIGNATURE QUIK)
+	include(Packaging)
+
+	# Fortran module dirs
+	# --------------------------------------------------------------------
+	set(XC_SERIAL_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/libxc/serial)
+	set(XC_MPI_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/libxc/mpi)
+	set(XC_GPU_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/libxc/gpu)
+
+	set(QUICK_SERIAL_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/quick/serial)
+	set(QUICK_MPI_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/quick/mpi)
+	set(QUICK_GPU_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/quick/gpu)
+	set(QUICK_MPI_GPU_MOD_DIR ${CMAKE_CURRENT_BINARY_DIR}/amber-modules/quick/mpi_gpu)
+
+endif()
+
+option(NOF "Disables the compilation of QUICK's time consuming f functions in the ERI code of cuda version. Not recommended for production." FALSE)
+
+# C/Fortran compiler flags
+# --------------------------------------------------------------------
+
+# add -DDEBUG in debug configuration
+set_property(DIRECTORY . PROPERTY COMPILE_DEFINITIONS $<$<CONFIG:Debug>:DEBUG>)
+
```

Comment:
Is there any difference between `.` and `${CMAKE_CURRENT_SOURCE_DIR}`? When I use `GLOB`, it seems that `./*.f90` doesn't work and I have to use `${CMAKE_CURRENT_SOURCE_DIR}/*.f90`

Reply:
You know, I'm not really sure, I think it kind of depends on the application.  In most cases in CMake paths are interpreted relative to the current source dir -- the only major exception is things like add_custom_command() where you specify a working directory.